src/main/webapp/discussion.js

```diff
@@ -103,53 +129,77 @@ async function fetchDiscussion() {
 }
 
 /**
- * Creates an element for displaying {@code comment}.
+ * Renders a comment and its replies, with a form to post a new reply.
  */
-function createComment(comment) {
-  const element = document.createElement('li');
-  element.setAttribute(ATTR_ID, comment.commentKey.id);
+class DiscussionComment extends HTMLElement {
+  constructor(comment) {
+    super();
+    this.attachShadow({mode: 'open'});
+    const shadow = TEMPLATE_COMMENT.content.cloneNode(true);
+    this.shadowRoot.appendChild(shadow);
+
+    const username = comment.author.email.split('@')[0];
+    let timestampPrefix = '';
+    if (!comment.parentKey.value) {
+      // Only display timestamp on root comments.
+      timestampPrefix = `${this.timestampToString(comment.timestamp)} - `;
+    }
+    const header = `${timestampPrefix}${username} on ${comment.created}`;
+    this.setSlotSpan(SLOT_HEADER, header);
```

Comment:
typically, constructors should never do any work. they should only build the object.

Reply:
You could move this logic to a method called "createFirstComment" or "initializeCommentForm" or some other method name depending on what you are doing.

Reply:
This could probably even be split into multiple methods. Seems like a lot is happening here.

Reply:
Will do